python/dgl/convert.py

```diff
@@ -912,31 +1109,253 @@ def _batcher(lst):
                     raise DGLError('Not all edges have attribute {}.'.format(attr))
             g.edata[attr] = F.copy_to(_batcher(attr_dict[attr]), g.device)
 
-    return g
+    return g.to(device)
+
+def bipartite_from_networkx(nx_graph,
+                            utype='_U', etype='_E', vtype='_V',
+                            src_attrs=None, edge_attrs=None, dst_attrs=None,
+                            edge_id_attr_name=None,
+                            idtype=None,
+                            device=None):
+    """Create a unidirectional bipartite graph from a NetworkX graph.
+
+    A bipartite graph has two types of nodes ``"SRC"`` and ``"DST"`` and
+    there are only edges between nodes of different types. By "unidirectional",
+    there are only edges from ``"SRC"`` nodes to ``"DST"`` nodes.
+
+    Creating a DGLGraph from a NetworkX graph is not fast especially for large scales.
+    It is recommended to first convert a NetworkX graph into a tuple of node-tensors
+    and then construct a DGLGraph with :func:`dgl.heterograph`.
+
+    Parameters
+    ----------
+    nx_graph : networkx.DiGraph
+        The NetworkX graph holding the graph structure and the node/edge attributes.
+        DGL will relabel the nodes using consecutive integers starting from zero if it is
+        not the case. The graph must follow `NetworkX's bipartite graph convention
+        <https://networkx.github.io/documentation/stable/reference/algorithms/bipartite.html>`_,
+        and furthermore the edges must be from nodes with attribute `bipartite=0` to nodes
+        with attribute `bipartite=1`.
+    utype : str, optional
+        The name of the source node type.
+    etype : str, optional
+        The name of the edge type.
+    vtype : str, optional
+        The name of the destination node type.
+    src_attrs : list[str], optional
+        The names of the ``"SRC"`` node attributes to retrieve from the NetworkX graph. If given,
+        DGL stores the retrieved node attributes in ``srcdata`` of the returned graph using their
```

Comment:
Why does `srcdata` come in here?  Can we say ``nodes[utype].data``?

Reply:
Done.